filebeat/input/filestream/internal/task/group.go

```diff
@@ -0,0 +1,129 @@
+package task
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"golang.org/x/sync/semaphore"
+)
+
+type Group struct {
+	limit uint64
```

Comment:
Nit: I think you can get rid of this field as the functionality we need from it is duplicated by the `sem` field.  If `sem` == `nil`, it's the same as `limit` being `0`.  And otherwise, we will use `sem` anyway to stay within the limit.  I think it's better to have just a single way to achieve the same functionality in the code to make it less confusing and also potentially less error-prone.